diff --git a/config.sub b/config.sub
index 38f3d037a..6a19e8b26 100755
--- a/config.sub
+++ b/config.sub
@@ -1749,7 +1749,7 @@ case $os in
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | roanix* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff --git a/fixincludes/mkfixinc.sh b/fixincludes/mkfixinc.sh
index df90720b7..b7f7a15a1 100755
--- a/fixincludes/mkfixinc.sh
+++ b/fixincludes/mkfixinc.sh
@@ -11,6 +11,8 @@ target=fixinc.sh
 
 # Check for special fix rules for particular targets
 case $machine in
+    *-roanix* | \
+    *-*-roanix* | \
     i?86-*-cygwin* | \
     i?86-*-mingw32* | \
     x86_64-*-mingw32* | \
diff --git a/gcc/config.gcc b/gcc/config.gcc
index a3566f5c7..5265af4d1 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -825,6 +825,18 @@ case ${target} in
   rust_target_objs="${rust_target_objs} dragonfly-rust.o"
   target_has_targetrustm=yes
   ;;
+*-*-roanix*)
+  extra_options="$extra_options gnu-user.opt"
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  use_gcc_stdint=provide
+  case ${enable_threads} in
+    "" | yes | posix)
+      thread_file='posix'
+      ;;
+  esac
+  ;;
 *-*-freebsd*)
   # This is the generic ELF configuration of FreeBSD.  Later
   # machine-specific sections may refine and add to this
@@ -1189,6 +1201,9 @@ case ${target} in
 esac
 
 case ${target} in
+x86_64-*-roanix*)
+	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h gnu-user.h glibc-stdint.h i386/i386elf.h i386/x86-64.h roanix.h"
+	;;
 aarch64*-*-elf | aarch64*-*-fuchsia* | aarch64*-*-rtems*)
 	tm_file="${tm_file} elfos.h newlib-stdint.h"
 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-errata.h aarch64/aarch64-elf-raw.h"
@@ -2469,6 +2484,25 @@ microblaze*-*-rtems*)
 	tmake_file="${tmake_file} microblaze/t-microblaze"
 	tmake_file="${tmake_file} microblaze/t-rtems"
         ;;
+riscv*-*-roanix*)
+	tm_file="elfos.h gnu-user.h glibc-stdint.h ${tm_file} riscv/elf.h roanix.h"
+	tmake_file="${tmake_file} riscv/t-riscv"
+	gnu_ld=yes
+	gas=yes
+	case $target in
+	riscv32be-*|riscv64be-*)
+		tm_defines="${tm_defines} TARGET_BIG_ENDIAN_DEFAULT=1"
+		;;
+	esac
+	gcc_cv_initfini_array=yes
+	case ${with_abi} in
+	lp64 | lp64f | lp64d)
+		;;
+	*)
+		with_abi=lp64d
+		;;
+	esac
+	;;
 microblaze*-*-elf)
 	case $target in
 		microblazeel-*)
@@ -3653,7 +3687,7 @@ case ${target} in
         ;;
 *-*-kfreebsd*-gnu | *-*-kopensolaris*-gnu)
         ;;
-*-*-linux* | *-*-gnu*)
+*-*-linux* | *-*-gnu* | *-*-roanix*)
 	case ${target} in
 	aarch64*-* | arm*-* | i[34567]86-* | powerpc*-* | s390*-* | sparc*-* | x86_64-* | loongarch*-*)
 		default_gnu_indirect_function=yes
diff --git a/gcc/config/roanix.h b/gcc/config/roanix.h
new file mode 100644
index 000000000..27ae883ce
--- /dev/null
+++ b/gcc/config/roanix.h
@@ -0,0 +1,21 @@
+#undef LIB_SPEC
+#define LIB_SPEC "-lc"
+
+#undef LINK_SPEC
+#define LINK_SPEC "%{shared:-shared} %{static:-static} %{!shared:%{!static:%{rdynamic:-export-dynamic}}} %{!dynamic-linker:-dynamic-linker /usr/lib/ld.so}"
+
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "%{!shared:crt1.o%s} crti.o%s %{shared:crtbeginS.o%s;:crtbegin.o%s}"
+
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "%{shared:crtendS.o%s;:crtend.o%s} crtn.o%s"
+
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS() \
+  do { \
+    builtin_define ("__roanix__"); \
+    builtin_define ("__unix__"); \
+    builtin_assert ("system=roanix"); \
+    builtin_assert ("system=unix"); \
+    builtin_assert ("system=posix"); \
+  } while (0)
diff --git a/libgcc/config.host b/libgcc/config.host
index ee10e99b0..497c8c53b 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -784,6 +784,10 @@ i[34567]86-*-openbsd*)
 	;;
 x86_64-*-openbsd*)
 	;;
+x86_64-*-roanix*)
+	extra_parts="$extra_parts crtbegin.o crtend.o crtbeginS.o crtendS.o crtprec32.o crtprec64.o crtprec80.o crtfastmath.o"
+	tmake_file="$tmake_file t-crtstuff-pic t-libgcc-pic i386/t-crtpc t-crtfm i386/t-crtstuff t-dfprules"
+	;;
 i[34567]86-*-linux*)
 	extra_parts="$extra_parts crtprec32.o crtprec64.o crtprec80.o crtfastmath.o"
 	tmake_file="${tmake_file} i386/t-crtpc t-crtfm i386/t-crtstuff t-dfprules"
@@ -1358,6 +1362,10 @@ riscv*-*-linux*)
 	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o crtendS.o crtbeginT.o"
 	md_unwind_header=riscv/linux-unwind.h
 	;;
+riscv*-*-roanix*)
+	tmake_file="${tmake_file} riscv/t-crtstuff riscv/t-softfp${host_address} t-softfp riscv/t-elf riscv/t-elf${host_address} t-slibgcc-libgcc"
+	extra_parts="$extra_parts crtbegin.o crtbeginS.o crtend.o crti.o crtn.o crtendS.o crtbeginT.o"
+	;;
 riscv*-*-freebsd*)
 	tmake_file="${tmake_file} riscv/t-crtstuff riscv/t-softfp${host_address} t-softfp riscv/t-elf riscv/t-elf${host_address} t-slibgcc-libgcc"
 	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o crtendS.o crtbeginT.o"
